packages/ra-core/src/controller/RecordContext.tsx

```diff
@@ -0,0 +1,63 @@
+import { createContext, useContext } from 'react';
+import pick from 'lodash/pick';
+import { Record } from '../types';
+
+export interface RecordContextValue {
+    record?: Partial<Record>;
+    loaded: boolean;
+    loading: boolean;
+}
+
+/**
+ * Context to store the result of the useRecord() hook.
+ *
+ * Use the useRecordContext() hook to read the context. That's what the Edit components do in react-admn.
+ *
+ * @example
+ *
+ * import { useEditController, EditContext } from 'ra-core';
+ *
+ * const Edit = props => {
+ *     const controllerProps = useEditController(props);
+ *     return (
+ *         <EditContext.Provider value={controllerProps}>
+ *             ...
+ *         </EditContext.Provider>
+ *     );
+ * };
+ */
+export const RecordContext = createContext<RecordContextValue>({
+    loaded: null,
+    loading: null,
+    record: null,
+});
+
+RecordContext.displayName = 'RecordContext';
+
```

Comment:
you should also expose a RecordContextProvider, to let us change the internal storage in the future without breaking compatibility

Reply:
Did it locally already